scopesim/effects/fits_headers.py

```diff
@@ -340,17 +340,17 @@ def flatten_dict(dic, base_key="", flat_dict=None,
                 value = deepcopy(val)
 
             # resolve any bang or hash strings
-            if resolve and isinstance(value, str) and len(value) > 0:
-                if value[0] == "!":
+            if resolve and isinstance(value, str) and value:
```

Comment:
```suggestion
            if resolve and isinstance(value, str):
```
I believe the `len(value) > 0` was only there to be able to do `value[0]`, and is thus redundant due to the use of `startswith`.